src/AwsLambda/AwsLambdaOpenTracer/DataCollector.cs

```diff
@@ -63,23 +65,43 @@ public void TransformSpans(LambdaRootSpan rootSpan)
                 TransactionEvent txnEvent = new TransactionEvent(rootSpan);
                 var errorEvents = _errors.GetAndClearEvents();
                 var errorTraces = _errors.GetAndClearTraces();
-                WriteData(arn, _executionEnv, spans.ToList(), txnEvent, errorEvents, errorTraces);
+
+                var (payload, data) = PreparePayload(arn, _executionEnv, spans.ToList(), txnEvent, errorEvents, errorTraces);
+
+                // If named pipe exists we want to send the payload there instead of writing to standard out.
+                if (File.Exists(_namedPipePath))
+                {
+                    WriteToNamedPipe(payload);
+                    return;
+                }
+                WriteData(payload, data);
             }
         }
 
-        // Write all the payload data to the console using standard out. This is the only method that should call the Logger#out method.
-        private void WriteData(string arn, string executionEnv, IList<Event> spans, TransactionEvent txnEvent, IList<ErrorEvent> errorEvents, IList<ErrorTrace> errorTraces)
+        private (string, IDictionary<string, object>) PreparePayload(string arn, string executionEnv, IList<Event> spans, TransactionEvent txnEvent, IList<ErrorEvent> errorEvents, IList<ErrorTrace> errorTraces)
```

Comment:
We don't typically use the new Tuple syntax in our code since we don't often deal with with, but this looks nice and easy to read.